wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt

```diff
@@ -159,6 +159,75 @@ class WirePluginTest {
         .contains("src/test/projects/sourcetree-many-srcdirs/build/generated/src/main/java")
   }
 
+  @Test
+  fun sourceJarLocalOneJarMultipleFiles() {
+    val fixtureRoot = File("src/test/projects/sourcejar-local-many-files")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+        .contains(
+            "src/test/projects/sourcejar-local-many-files/build/generated/src/main/java"
+        )
+  }
+
+  @Test
+  fun sourceJarLocalOneJarSingleFile() {
+    val fixtureRoot = File("src/test/projects/sourcejar-local-single-file")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .withDebug(true)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .doesNotContain("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+        .contains(
+            "src/test/projects/sourcejar-local-single-file/build/generated/src/main/java"
+        )
+  }
+
+  @Test
+  fun sourceJarRemoteOneJarMultipleFiles() {
+    val fixtureRoot = File("src/test/projects/sourcejar-remote-many-files")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+        .contains(
+            "src/test/projects/sourcejar-remote-many-files/build/generated/src/main/java"
+        )
+  }
+
+  @Test
+  fun sourceJarRemoteOneJarWithProtoPath() {
+    val fixtureRoot = File("src/test/projects/sourcejar-remote-protopath")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
```

Comment:
`period.proto` is on the `protoPath`, doesn't it mean that it shouldn't get compiled, as it's a dependency?

Reply:
great catch.  turns out the test setup isn't quite right.  i'll fix that.

Reply:
https://github.com/square/wire/pull/940

Reply:
fixed